test_conformance/images/clFillImage/test_fill_generic.cpp

```diff
@@ -83,27 +103,7 @@ cl_mem create_image( cl_context context, cl_command_queue queue, BufferOwningPtr
                 cl_mem_flags buffer_flags = CL_MEM_READ_WRITE;
                 if (gEnablePitch)
                 {
-                    cl_device_id device;
-                    err =
-                        clGetCommandQueueInfo(queue, CL_QUEUE_DEVICE,
-                                              sizeof(device), &device, nullptr);
-                    if (err != CL_SUCCESS)
-                    {
-                        log_error(
-                            "Error: Could not get CL_QUEUE_DEVICE from queue");
-                        return NULL;
-                    }
-                    char major_version;
-                    err = clGetDeviceInfo(device, CL_DEVICE_VERSION,
-                                          sizeof(major_version), &major_version,
-                                          nullptr);
-                    if (err != CL_SUCCESS)
-                    {
-                        log_error("Error: Could not get CL_DEVICE_VERSION from "
-                                  "device");
-                        return NULL;
-                    }
-                    if (major_version == '1')
+                    if (CL_VERSION_MAJOR(version) == 1)
```

Comment:
- If major version is not 1 and image type is other than IMAGE1D_BUFFER this code will call align_free instead of free